shared/src/main/scala/mlscript/ConstraintSolver.scala

```diff
@@ -759,6 +766,7 @@ class ConstraintSolver extends NormalForms { self: Typer =>
           case (p @ ProvType(und), _) => rec(und, rhs, true)
           case (_, p @ ProvType(und)) => rec(lhs, und, true)
           case (_: TypeTag, _: TypeTag) if lhs === rhs => ()
+          case (lhs: Unsupported, rhs: Unsupported) => ()
```

Comment:
That looks very fishy. Do you have an example?

Reply:
```ts
interface ErrorConstructor {
    new(message?: string): Error;
    (message?: string): Error;
    readonly prototype: Error;
}

interface EvalErrorConstructor extends ErrorConstructor {
    new(message?: string): EvalError;
    (message?: string): EvalError;
    readonly prototype: EvalError;
}
```

In this case, `new(message?: string): Error` and `new(message?: string): EvalError` are construction signatures and not supported. Since `EvalErrorConstructor` extends `ErrorConstructor`, the constraint solver will try to solve there two construction signatures, which leads to type errors in `es5.mlsi`. So here I marked that two `Unsupported` types are always the same. But you are right, maybe we can apply a better solution in `ts2mls`.

Reply:
Ok that's fine, but explain why this is needed in a comment, ie "required to allow signature overriding checks in declarations"